src/ng1/directives/spark/spark.spec.js

```diff
@@ -33,40 +33,101 @@ describe('spark charts', function () {
+      });
+
+      it("should properly display the top right", function() {
+        expect(element2.find('.align-right .spark-label')[0].textContent).toBe('75.0M');
+      });
+
+      it("should properly display the bottom left", function() {
+        expect(element2.find('.spark-label')[1].textContent).toBe("ITEMS ON HOLD");
+      });
+
+      it("should properly display the bottom right", function() {
+        expect(element2.find('.align-right .spark-label')[1].textContent).toBe('TOTAL');
+      });
+
+      it("should properly display the inline label", function () {
+        expect(element.find('.spark-label-left')[0].textContent.trim()).toBe('60%');
+      });
     });
-    it('should display the spark bottom-right label with proper values', function () {
-      expect(element2.find('.align-right .spark-label')[1].textContent).toBe('TOTAL');
+
+    describe("when the fillheight property is set", function (){
+      it("should properly set the spark height", function () {
+        expect(element.find('.spark-primary').css('height')).toBe('5px');
+      });
     });
-    it('should set aria-valuenow to the value passed', function () {
-      expect(element.find('.spark-primary > div.progress-bar').attr("aria-valuenow")).toBe('35');
-      expect(element2.find('.spark-secondary3 > div.progress-bar').attr("aria-valuenow")).toBe('30');
+
+    describe("when all the colors are set", function (){
+      it("should properly set the bar colors for each segment", function() {
+        expect(multiValueSpark.find(".spark .progress-bar").eq(0).css("background-color")).toBe("rgb(26, 172, 96)");
+        expect(multiValueSpark.find(".spark .progress-bar").eq(1).css("background-color")).toBe("rgb(252, 219, 31)");
+        expect(multiValueSpark.find(".spark .progress-bar").eq(2).css("background-color")).toBe("rgb(229, 0, 76)");
+      });
+
+      it("should properly set the track color", function() {
+        expect(element2.find("." + vm.type2).css("background-color")).toBe("rgb(255, 0, 0)");
+      });
     });
-    it('should display the inline label properly', function () {
-      expect(element.find('.spark-label-left')[0].textContent.trim()).toBe('60%');
+
+    describe("when tooltips are set in a multi-value spark", function (){
+      it("should properly set the bar tooltips for each segment", function() {
+        expect(multiValueSpark.find(".spark .progress-bar").eq(0).attr("tooltip")).toBe(vm.tooltips[0]);
+        expect(multiValueSpark.find(".spark .progress-bar").eq(1).attr("tooltip")).toBe(vm.tooltips[1]);
+        expect(multiValueSpark.find(".spark .progress-bar").eq(2).attr("tooltip")).toBe(vm.tooltips[2]);
+      });
     });
-    it('should add proper height according to the value of fillheight', function () {
-      expect(element.find('.spark-primary').css('height')).toBe('5px');
+
+    describe("when segment click is set and clicked in a multi-value spark", function (){
```

Comment:
This is the test failing for me, Ashley/Donna. The markup for `multiValueSpark` is not including the `segment-click` property, not sure why. Can you take a look at the initialization and let me know if there's anything I should be doing differently?

Reply:
Hi @victor-ac, I have pushed a few changes, the tests now seem to be passing. Thanks for the contribution!

Reply:
@donna-marijne requesting review as I have made additional changes